File: 0001.rb
def build_array size
  Array 1..size
end

def add_up_multiples numbers
  numbers.reduce(0) do |total, num|
    (num % 3 == 0 or num % 5 == 0) ? total + num : total
  end
end

def solve number
  number_array = build_array(number - 1)
  add_up_multiples(number_array)
end

puts solve 1000

File: 0001.js
let buildArray = size => {
  let numbers = new Array(size);
  for (let i = 0; i < numbers.length; i++) {
    numbers[i] = i + 1;
  }
  return numbers;
}

let addUpMultiples = numbers => {
  return numbers.reduce((prev, next) => {
    if (next % 3 === 0 || next % 5 === 0) {
      return prev + next;
    }
    return prev;
  }, 0);
}

let solve = number => {
  const numberArray = buildArray(number - 1);
  const total = addUpMultiples(numberArray);
  return total;
}

console.log(solve(1000));